File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        char[] sArray = s.toLowerCase().toCharArray();
        int left = 0;
        int right = sArray.length - 1;
        while(left < right){
            
            if(!Character.isLetterOrDigit(sArray[left]) ){
                left++;
                continue;
            }
            
            if(!Character.isLetterOrDigit(sArray[right]) ){
                right--;
                continue;
            }
            
            if(sArray[left] != sArray[right]) return false;
            
            left++;
            right--;
        }
        
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.kt
class Solution {
    fun isPalindrome(s: String): Boolean {

        val stack = ArrayDeque<Char>()

        for(char in s){
            if(!char.isLetterOrDigit()) continue
            stack.addLast(char.toLowerCase())
        }

        for(i in 0 until s.length){
            var current = s[i].toLowerCase()
            if(!current.isLetterOrDigit()) continue
            if(stack.removeLast() != current) return false
        }
        if(stack.isNotEmpty()) return false

        return true
    }
}